### PR TITLE
[ACA-3407] New component : Filter Menu created and implemented inside the Document List

### PR DESCRIPTION
**Pull request by BaptisteMahe**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
No content is implemented inside the DataTable Header through the Document List.

**What is the new behaviour?**
New component created: FilterMenuComponent. This component will be used to display and control the filters inside the DocumentList.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ACA-3407
Screenshot 2020-06-05 at 18 53 38

### CODE REVIEW

**Comment by DenysVuika** [comment]
Can this be a regular vertical menu items?

File: lib/content-services/src/lib/document-list/components/filter-menu/filter-menu.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<button mat-icon-button [matMenuTriggerFor]="filter"
+    (click)="stopEventPropagation($event)"
```
Comment (VitoAlbano) [inline lib/content-services/src/lib/document-list/components/filter-menu/filter-menu.component.html:2]: as a policy we agreed to call the method related to the event with 'on'TypeOfEvent in this case i expect this to be OnClick or if you want to be more detailed with the method name like : onButtonMenuClick(

**Comment by BaptisteMahe** [comment]
> Can this be a regular vertical menu items?

What do you mean by "regular vertical items"?

**Comment by DenysVuika** [comment]
You are putting dialog actions in the pop up. This is not a dialog from what I understand, so should look like a menu with items stacked vertically. Like. Context menu or document actions menu.

**Comment by BaptisteMahe** [comment]
> You are putting dialog actions in the pop up. This is not a dialog from what I understand, so should look like a menu with items stacked vertically. Like. Context menu or document actions menu.

We are trying to meet this specification, do you think that it should be changed/reviewed? :
Screenshot 2020-06-03 at 16 03 31

**Comment by DenysVuika** [comment]
Can we then have the end result look closer to the spec?

**Comment by BaptisteMahe** [comment]
> Can we then have the end result look closer to the spec?

Yes yes, of course, this is just the just a starting point. The search widget isn't implemented and styles aren't updated yet.

**Pull request merged**